pages/index.tsx

```diff
@@ -1,65 +1,44 @@
-import Head from "next/head";
-import styles from "../styles/Home.module.css";
+import { useSession } from "next-auth/client";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+import { UserRoleType } from "interfaces";
+import { UserDTO } from "./api/admin/users/readOneEmail";
 
 const Home: React.FC = () => {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
-
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
-
-        <p className={styles.description}>
-          Get started by editing{" "}
-          <code className={styles.code}>pages/index.js</code>
-        </p>
-
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h3>Documentation &rarr;</h3>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+  const [session, loading] = useSession();
+  const router = useRouter();
+
+  useEffect(() => {
+    const fetchUserRole = async (): Promise<void> => {
+      if (session) {
+        const response = await fetch("/api/admin/users/readOneEmail", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          credentials: "include",
+          body: JSON.stringify({
+            email: session.user.email,
+          } as UserDTO),
+        });
+        const data = await response.json();
+        if (!response.ok) {
+          router.push("/auth/signin");
+        }
+        const role = data?.user.roles[0].type;
```

Comment:
this line seems to throw and error when you try to sign in as anything other than an admin user